ui/packages/catalog-ui-search/src/main/webapp/component/alert/alert.js

```diff
@@ -68,6 +71,46 @@ define([
             this.listenTo(this, 'change:currentAlert', this.clearSelectedResults);
             this.listenTo(this.get('activeSearchResults'), 'update add remove reset', this.updateActiveSearchResultsAttributes);
             this.listenTo(this.get('completeActiveSearchResults'), 'update add remove reset', this.updateActiveSearchResultsFullAttributes);
+            this.listenTo(router, 'change', this.handleRoute);
+            this.handleRoute();
+        },
+        handleRoute() {
+            const routerJSON = router.toJSON();
+            if (routerJSON.name === 'openAlert'){
+                var alertId = routerJSON.args[0];
+                var alert = user.get('user').get('preferences').get('alerts').get(alertId);
+                if (!alert) {
+                    router.notFound();
+                } else {
+                    const queryForMetacards = new Query.Model({
+                        cql: cql.write({
+                            type: 'OR',
+                            filters: alert.get('metacardIds').map(function(metacardId){
+                                return {
+                                    type: '=',
+                                    value: metacardId,
+                                    property: '"id"'
+                                };
+                            }).concat({
+                                type: '=',
+                                value: '-1',
+                                property: '"id"'
+                            })
+                        }),
+                        federation: 'enterprise'
```

Comment:
❓ from a UI perspective, how often is an enterprise query issued to the backend?  if it's any time a metacard is accessed, then i see this as an issue

Reply:
It's only when a user routes to an alert ('alert/12323212312312').  We only have a collection of ids at that point, so we have to do an enterprise query in an attempt to resolve them.

This code was originally located here: https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/js/router.js#L174

In the future we might be able to avoid an enterprise query if we query locally first, check to see if all the ids are resolved, and if they are entirely resolved we can stop there.  If they aren't, we can move on to querying the enterprise.